penman/types.py

```diff
@@ -2,14 +2,15 @@
 Basic types used by various Penman modules.
 """
 
-from typing import Any, Iterable, List, Tuple, Union
+from typing import Iterable, List, Tuple, Union
 
 Variable = str
 Constant = Union[str, float, int, None]  # None for missing values
 Role = str  # '' for anonymous relations
+Symbol = str
```

Comment:
Is this the correct name for what's in branch targets? It seemed like the target must always be a string if it's not a `Node`, with even number constants showing up as strings. Are there edge-cases of trees where this isn't correct?

Reply:
> Is this the correct name for what's in branch targets?

I believe this would be `Atom`. Considering the [grammar](https://penman.readthedocs.io/en/latest/notation.html#formal-grammar), atoms are variables or constants, constants are either strings or symbols. Here's an example:

```lisp
(a / A
   :ARG1 (b / B)  ; node target
   :ARG2 b        ; variable target (atomic)
   :ARG3 "a b"    ; string target   (atomic)
   :ARG4 abc      ; symbol target   (atomic)
)
```

The only difference between a symbol and a variable is that a variable is used in a node (such as `a` and `b` in the example above). The difference between strings and symbols, besides the quotes, is just that symbols cannot contain control characters like whitespace, parens, colons, etc. Quoted strings may not be used as variables (`("a" / A)` raises an error, and `:ARG1 "a"` does not reference a variable `a`).

> even number constants showing up as strings

That is correct. Penman does no interpretation of datatypes on parsing. It will accept a limited number of non-string types (ints, floats, etc.) during encoding, but they will be strings again when decoding.

> Are there edge-cases of trees where this isn't correct?

While unconventional, missing targets are allowed and parse to `None` (a warning may be issued as well):

```pycon
>>> penman.parse('(a / A :ARG1)')
Missing target: (a / A :ARG1)
Tree(('a', [('/', 'A'), (':ARG1', None)]))
```

Similarly, an empty node target means that the type annotation of `Node` is not entirely correct, as mentioned in #129:

```pycon
>>> penman.parse('(a / A :ARG1 ())')
Tree(('a', [('/', 'A'), (':ARG1', (None, []))]))
```